lib/kinesis_client/stream/shard/producer.ex

```diff
@@ -131,49 +135,29 @@ defmodule KinesisClient.Stream.Shard.Producer do
   end
 
   @impl GenStage
-  def handle_info({:ack, _ref, [], failed_msgs}, state) do
```

Comment:
Doesn't this mean we just removed all retries from the stream consumer though too? Are we saying we absolutely never want this behavior?

Reply:
That's correct, because it gets us into retry loops like we saw today where we never advance. Retries make sense in SQS or Rabbit land, but Kafka, Kinesis, and other streams should move on. If we do want to really retry, we can implement the `handle_failed` callback.